network/p2p/gossip/gossip.go

```diff
@@ -76,6 +120,10 @@ var (
 	ErrInvalidDiscardedSize     = errors.New("discarded size cannot be negative")
 	ErrInvalidTargetGossipSize  = errors.New("target gossip size cannot be negative")
 	ErrInvalidRegossipFrequency = errors.New("re-gossip frequency cannot be negative")
+	// ErrGossipableAlreadyKnown is returned by the Set's Add method if the gossipable is already present.
+	ErrGossipableAlreadyKnown = errors.New("gossipable already known")
+	// ErrGossipableMalformed indicates a gossipable with invalid structure or content.
+	ErrGossipableMalformed = errors.New("gossipable malformed")
```

Comment:
Instead of defining drop reasons + metrics in the `gossip` package, should we define them in the vm's mempool implementation? One example where this is awkward is we're using `ErrGossipableMalformed` when failing verification in the mempool, when this error feels more like an error intended for parsing failures.

I think one way we could do this is to have a mempool implementation in this package that handles dropping duplicate/txs that fail to parse, that is then wrapped by the vm mempool implementation which adds its own metrics.

Reply:
Yes, I was thinking the same. The reason I held back was that it seems to be a bit "too much" for what we're trying to achieve. I did update the labels to separate the Malformed ( parsing error ) and FailedVerification ( for verification failure ) as @StephenButtolph suggested.
i.e. the goal in this PR is to be able to identify if we have any of these fail cases at all.

Please LMK if you'd want to proceed with the code as is or the refactor as you suggested above.

Reply:
I think refactoring would be better - otherwise I think the point of this PR will be lost (increasing visibility) if we bucket unrelated drop reasons into things like malformed/other IMO.